tools/lib/url_file.py

```diff
@@ -35,13 +36,14 @@ def __init__(self, url, debug=False, cache=None):
     if cache is not None:
       self._force_download = not cache
 
-    try:
-      self._curl = self._tlocal.curl
-    except AttributeError:
-      self._curl = self._tlocal.curl = pycurl.Curl()
     if not self._force_download:
       os.makedirs(Paths.download_cache_root(), exist_ok=True)
 
+  def _get_http_client(self):
```

Comment:
wrong indentation again. can you see why the linter isn't catching this?

Reply:
Changing ruff to..
```
[tool.ruff]
preview = true
select = ["E", "F", "W", "PIE", "C4", "ISC", "RUF008", "RUF100", "A", "B", "TID251", "E117"]
```
should be what were [looking for](https://docs.astral.sh/ruff/rules/over-indented/) but we'd need to set the [indent width](https://docs.astral.sh/ruff/settings/#indent-width) to 2 which wasn't working for me
```
E111 Indentation is not a multiple of 4
```

Reply:
can we turn off E111?

Reply:
preview
```
preview = true
Found 17014 errors.
```
```
preview = true
ignore = [.. "E111"]
Found 2630 errors.
```

no preview
```
(overindented - 4 spaces)
ignore = [.. "E111"]
Passed
```
```
(overindented - over 4 spaces)
ignore = [.. "E111"]
Found 1 errors. E117 Over-indented
```

adding this did not make a difference
```
indent-width = 2
```

Reply:
ah seems like it's not ready in ruff yet? that's fine if that's the case